components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -877,6 +878,166 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
+            changes must be applied via Group Resource.
+             */
+            if (claims.containsKey(SCIMConstants.UserSchemaConstants.ROLES_URI + "." + SCIMConstants.DEFAULT)) {
+                claims.remove(SCIMConstants.UserSchemaConstants.ROLES_URI);
+            }
+
+            claims.remove(SCIMConstants.UserSchemaConstants.USER_NAME_URI);
+
+            /*
+            Since we are already populating last_modified claim value from SCIMUserOperationListener, we need to
+            remove this claim value which is coming from charon-level.
+             */
+            claims.remove(SCIMConstants.CommonSchemaConstants.LAST_MODIFIED_URI);
+
+            // Location is a meta attribute of user object.
+            claims.remove(SCIMConstants.CommonSchemaConstants.LOCATION_URI);
+
+            // Resource-Type is a meta attribute of user object.
+            claims.remove(SCIMConstants.CommonSchemaConstants.RESOURCE_TYPE_URI);
+
+            Map<String, String> scimToLocalClaimsMap = SCIMCommonUtils.getSCIMtoLocalMappings();
+            List<String> requiredClaims = getOnlyRequiredClaims(scimToLocalClaimsMap.keySet(), requiredAttributes);
+            List<String> requiredClaimsInLocalDialect;
+            if (MapUtils.isNotEmpty(scimToLocalClaimsMap)) {
+                scimToLocalClaimsMap.keySet().retainAll(requiredClaims);
+                requiredClaimsInLocalDialect = new ArrayList<>(scimToLocalClaimsMap.values());
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("SCIM to Local Claim mappings list is empty.");
+                }
+                requiredClaimsInLocalDialect = new ArrayList<>();
+            }
+
+            // Get existing user claims.
+            Map<String, String> oldClaimList = carbonUM.getUserClaimValuesWithID(user.getId(),
+                    requiredClaimsInLocalDialect.toArray(new String[0]), null);
+
+            oldClaimList.remove(LOCATION_CLAIM);
+            oldClaimList.remove(LAST_MODIFIED_CLAIM);
+            oldClaimList.remove(RESOURCE_TYPE_CLAIM);
+
+            // Get user claims mapped from SCIM dialect to WSO2 dialect.
+            Map<String, String> claimValuesInLocalDialect = SCIMCommonUtils.convertSCIMtoLocalDialect(claims);
+
+            // If password is updated, set it separately.
+            if (user.getPassword() != null) {
+                carbonUM.updateCredentialByAdminWithID(user.getId(), user.getPassword());
+            }
+
+            updateUserClaims(user, oldClaimList, claimValuesInLocalDialect, allSimpleMultiValuedClaimsList);
+
+            if (log.isDebugEnabled()) {
+                log.debug("User: " + user.getUserName() + " updated through SCIM.");
+            }
+            return getUser(user.getId(), requiredAttributes);
+        } catch (UserStoreException e) {
+            log.error("Error while updating attributes of user: " + user.getUserName(), e);
```

Comment:
Avoid logging the error and throwing it the same time.

Reply:
Shall we fix it in other places?

Reply:
addressed in https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/266/commits/d419b5d649f8bc9093b6c4143d0fa188cce5ad99